### PR TITLE
Move monitoring to odh-manifests

### PR DESCRIPTION
**Pull request by mroman-redhat**
Moving monitoring components to odh-manifests to solve jira issue 701

### CODE REVIEW

File: deploy.sh [hunk 1]
```diff
@@ -98,10 +98,10 @@ if [ $? -ne 0 ]; then
   exit 1
 fi
 
+# Should this stay here if monitoring got removed?
 oc new-project $ODH_MONITORING_PROJECT || echo "INFO: $ODH_MONITORING_PROJECT project already exists."
```
Comment (anishasthana) [inline deploy.sh:102]: We will still need the project creation

**Review by anishasthana — APPROVED** [review]
This set of changes looks good. I vote we update the odh-manifests PR as well and then merge this in.

File: rhods-monitoring.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  - kustomizeConfig:
+      repoRef:
+        name: manifests
+        path: monitoring
+    name: monitoring
+  repos:
+  - name: manifests
+    uri: https://github.com/mroman-redhat/odh-manifests/tarball/move-monitoring
+  version: v1.0.0
+
```
Comment (crobby) [inline rhods-monitoring.yaml:17]: Extra newline here??

**Comment by crobby** [comment]
I think this looks pretty good.  As part of this, we probably need some way to either communicate to QE or provide PRs to their test repos for the cleanup of the monitoring KfDef.  (I will also need to update my script)
We might also need to update some of our documentations to reflect the new reality.

**Review by crobby — APPROVED** [review]

File: rhods-monitoring.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+apiVersion: kfdef.apps.kubeflow.org/v1
+kind: KfDef
+metadata:
+  name: redhat-ods-monitoring
```
Comment (crobby) [inline rhods-monitoring.yaml:4]: Maybe just "monitoring" so the object name isn't confused with the namespace??

File: rhods-monitoring.yaml [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+spec:
+  applications:
+  - kustomizeConfig:
+      repoRef:
+        name: manifests
+        path: monitoring
+    name: monitoring
+  repos:
+  - name: manifests
+    uri: https://github.com/mroman-redhat/odh-manifests/tarball/move-monitoring
```
Comment (crobby) [inline rhods-monitoring.yaml:15]: this also needs to change before we can merge

**Review by crobby — CHANGES_REQUESTED** [review]
need to reference the "local" manifests in the operator

**Review by anishasthana — COMMENTED** [review]
Changes look good but I haven't been able to test it out yet. @mroman-redhat  is this deployed anywhere?

**Comment by mroman-redhat** [comment]
> Changes look good but I haven't been able to test it out yet. @mroman-redhat is this deployed anywhere?

This should be it https://console-openshift-console.apps.mroman.dev.datahub.redhat.com/k8s/ns/redhat-ods-operator/pods

**Comment by anishasthana** [comment]
Hmm, I deleted the `rhods-rules` prometheus rule but the operator still hasn't recreated it.

**Review by crobby — APPROVED** [review]

**Review by crobby — APPROVED** [review]
I'm good with these changes.

**Comment by crobby** [comment]
> Hmm, I deleted the `rhods-rules` prometheus rule but the operator still hasn't recreated it.

Is this behavior different from the experience before?

**Comment by anishasthana** [comment]
Weirdly enough, it seems like it _did_ eventually reconcile. It just took something like an hour to do so.

**Pull request merged**